Src/ILGPU/Runtime/OpenCL/CLAccelerator.cs

```diff
@@ -527,12 +527,25 @@ protected override CLKernel CreateKernel(CLCompiledKernel compiledKernel)
             new CLKernel(this, compiledKernel, launcher);
 
         /// <summary cref="Accelerator.CreateStream"/>
-        protected override AcceleratorStream CreateStreamInternal() =>
-            new CLStream(this);
+        protected override AcceleratorStream CreateStreamInternal()
+        {
+            var stream = new CLStream(this);
+            ChildObjects.Add(new WeakReference<AcceleratorObject>(stream));
+            return stream;
+        }
 
         /// <summary cref="Accelerator.Synchronize"/>
-        protected override void SynchronizeInternal() =>
-            DefaultStream.Synchronize();
+        protected override void SynchronizeInternal()
+        {
+            for(int i = 0; i < ChildObjects.Count; i++)
```

Comment:
ISSUE: Concurrency; this will not work as it is not using the internal synchronization lock.

Reply:
I see. I didn't understand that the GC was asynchronous, although I don't see why it would be otherwise anyways.